src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -78,13 +89,44 @@ class WalletService(
             .filter { it.status == WalletStatusDto.CREATED }
             .switchIfEmpty { Mono.error(WalletConflictStatusException(WalletId(walletId))) }
             .flatMap {
+                val orderId = UUID.randomUUID().toString().replace("-", "").substring(0, 15)
+                val customerId = UUID.randomUUID().toString().replace("-", "").substring(0, 15)
+                val basePath = URI.create(sessionUrlConfig.basePath)
+                val merchantUrl = sessionUrlConfig.basePath
+                val resultUrl = basePath.resolve(sessionUrlConfig.outcomeSuffix)
+                val cancelUrl = basePath.resolve(sessionUrlConfig.cancelSuffix)
+                val notificationUrl =
+                    UriComponentsBuilder.fromHttpUrl(sessionUrlConfig.notificationUrl)
+                        .build(
+                            Map.of("orderId", orderId, "paymentMethodId", it.paymentMethodId.value)
+                        )
+
                 npgClient
                     .createNpgOrderBuild(
                         UUID.randomUUID(),
                         CreateHostedOrderRequest()
-                            .version("2")
-                            .merchantUrl("https://test")
-                            .order(Order())
+                            .version(CREATE_HOSTED_ORDER_REQUEST_VERSION)
+                            .merchantUrl(merchantUrl)
+                            .order(
+                                Order()
+                                    .orderId(orderId)
+                                    .amount(CREATE_HOSTED_ORDER_REQUEST_VERIFY_AMOUNT)
+                                    .currency(CREATE_HOSTED_ORDER_REQUEST_CURRENCY_EUR)
+                                    .customerId(customerId)
+                            )
+                            .paymentSession(
+                                PaymentSession()
+                                    .actionType(ActionType.VERIFY)
+                                    .amount(CREATE_HOSTED_ORDER_REQUEST_VERIFY_AMOUNT)
+                                    .language(CREATE_HOSTED_ORDER_REQUEST_LANGUAGE_ITA)
+                                    .paymentService(
+                                        "CARDS"
+                                    ) // execute call for retrieve by payment method id the service
```

Comment:
TODO for another PR?

Reply:
Yes, I think it's too onerous to predict it for this sprint.

Reply:
yep